package-res/resources/web/pentaho/visual/base/types/selectionModes.js

```diff
@@ -0,0 +1,88 @@
+/*!
+ * Copyright 2010 - 2016 Pentaho Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+define(function() {
+  "use strict";
+
+  /**
+   * The `selectionModes` namespace contains a collection of methods that represent different ways
+   * to combine the current selection filter with the selection filter that was created by the user
+   * interaction with a visualization.
+   *
+   * @name selectionModes
+   * @namespace
+   * @memberOf pentaho.visual.base.types
+   */
+  var selectionModes = /** @lends pentaho.visual.base.types.selectionModes */{
+    /**
+     * Replaces the current selection filter with the provided filter.
+     *
+     * @param {pentaho.data.filter.AbstractFilter} current - The current selection filter.
+     * @param {pentaho.data.filter.AbstractFilter} input - The input filter.
+     * @return {!pentaho.data.filter.AbstractFilter} The input filter.
+     * @static
+     */
+    REPLACE: function(current, input) {
+      return input;
+    },
+
+    /**
+     * Adds the input filter to the current selection filter if
```

Comment:
@davidmsantos90 < just because that is fun
________________do we really need to repeat "current selection filter" twice in the same sentence?

 Adds the input filter to the current selection filter, if it is not already fully contained within the selection filter.
 Otherwise, removes the input filter from the current selection filter.